src/geometry/obb.rs

```diff
@@ -34,95 +55,66 @@ impl<S: BaseFloat> Obb<S> {
+    fn compute_corners(&self) -> [Point3<S>; 8] {
+        let corner_from = |x, y, z| &self.query_from_obb * &Point3::new(x, y, z);
         [
-            corner_from(-half_extent.x, -half_extent.y, -half_extent.z),
-            corner_from(half_extent.x, -half_extent.y, -half_extent.z),
-            corner_from(-half_extent.x, half_extent.y, -half_extent.z),
-            corner_from(half_extent.x, half_extent.y, -half_extent.z),
-            corner_from(-half_extent.x, -half_extent.y, half_extent.z),
-            corner_from(half_extent.x, -half_extent.y, half_extent.z),
-            corner_from(-half_extent.x, half_extent.y, half_extent.z),
-            corner_from(half_extent.x, half_extent.y, half_extent.z),
+            corner_from(
+                -self.half_extent.x,
+                -self.half_extent.y,
+                -self.half_extent.z,
+            ),
+            corner_from(self.half_extent.x, -self.half_extent.y, -self.half_extent.z),
+            corner_from(-self.half_extent.x, self.half_extent.y, -self.half_extent.z),
+            corner_from(self.half_extent.x, self.half_extent.y, -self.half_extent.z),
+            corner_from(-self.half_extent.x, -self.half_extent.y, self.half_extent.z),
+            corner_from(self.half_extent.x, -self.half_extent.y, self.half_extent.z),
+            corner_from(-self.half_extent.x, self.half_extent.y, self.half_extent.z),
+            corner_from(self.half_extent.x, self.half_extent.y, self.half_extent.z),
         ]
     }
 
-    fn precompute_separating_axes(query_from_obb: &Quaternion<S>) -> Vec<Vector3<S>> {
-        let unit_x = Vector3::unit_x();
-        let unit_y = Vector3::unit_y();
-        let unit_z = Vector3::unit_z();
-        let rot_x = query_from_obb * unit_x;
-        let rot_y = query_from_obb * unit_y;
-        let rot_z = query_from_obb * unit_z;
-        let mut separating_axes = vec![unit_x, unit_y, unit_z];
-        for axis in &[
-            rot_x,
-            rot_y,
-            rot_z,
-            unit_x.cross(rot_x).normalize(),
-            unit_x.cross(rot_y).normalize(),
-            unit_x.cross(rot_z).normalize(),
-            unit_y.cross(rot_x).normalize(),
-            unit_y.cross(rot_y).normalize(),
-            unit_y.cross(rot_z).normalize(),
-            unit_z.cross(rot_x).normalize(),
-            unit_z.cross(rot_y).normalize(),
-            unit_z.cross(rot_z).normalize(),
-        ] {
-            let is_finite_and_non_parallel = is_finite(&axis)
-                && separating_axes.iter().all(|elem| {
-                    (elem - axis).magnitude() > S::default_epsilon()
-                        && (elem + axis).magnitude() > S::default_epsilon()
-                });
-            if is_finite_and_non_parallel {
-                separating_axes.push(*axis);
-            }
+    fn intersector(&self) -> Intersector<S> {
+        let corners = self.compute_corners();
+        let edges = self.compute_edges();
```

Comment:
Nit: It's a matter of taste, but I would just get rid of `compute_edges`, as it is only used once...

Reply:
Done.